packages/editor-ui/src/stores/usage.ts

```diff
@@ -1,52 +1,131 @@
 import { computed, reactive } from 'vue';
 import { defineStore } from 'pinia';
 import { UsageState } from '@/Interface';
-import { getLicense } from '@/api/usage';
+import { activateLicenseKey, getLicense, renewLicense } from '@/api/usage';
 import { useRootStore } from '@/stores/n8nRootStore';
+import { useSettingsStore } from "@/stores/settings";
+import { useUsersStore } from "@/stores/users";
+import { i18n } from '@/plugins/i18n';
 
-export const useUsageStore = defineStore('usageAndPlan', () => {
-	const rootStore = useRootStore();
+export type UsageTelemetry = {
+	instance_id: string
+	action: 'view_plans' | 'manage_plan' | 'add_activation_key'
+	plan_name_current: string
+	usage: number
+	quota: number
+};
 
-	const state = reactive<UsageState>({
-		loading: true,
-		error: null,
-		data: {
-			usage: {
-				executions: {
-					limit: -1,
-					value: 0,
-					warningThreshold: .8,
-				},
-			},
-			license: {
-				planId: 'community',
-				planName: 'Community',
+const SUBSCRIPTION_APP_URL = 'https://staging-subscription.n8n.io';
+const DEFAULT_PLAN_NAME = 'Community';
+const DEFAULT_STATE: UsageState = {
```

Comment:
If I make `data` optional or `null` in the interface I would have to use optional chaining everywhere and/or writing conditions (also in the template)
I would be up for the solution we don't show anything in case of an error but some message, but we already have that popping-up notification but the latest correct state is still rendered in the background.
The design is not prepared to show "nothing".

What if we just set loading to false in the mounted method when the request was successful? With this we show nothing in the place of the content and just have the notification?
But I dont' know what would the user do in that case. Reload? Navigate to another route and come back?

At some point we have to render something anyway that can change later and every time the network request fails we just remove the content?

Reply:
if the request fails, we can keep the old stale content.. but a little stale content is better than false content.. as long as we never show false content..

Reply:
Ok, this will also get a lot uglier because of the computed props which depending on other computed props.

e.g.:
```ts
const executionLimit = computed(() => state.data.usage.executions.limit);
const executionCount = computed(() => state.data.usage.executions.value);
const executionPercentage = computed(() => (executionCount.value / executionLimit.value) * 100);
```

if `state.data` can be `null` I can forget about this nice code

Reply:
this is your call here.. as long as we never show false data, it's fine by me

Reply:
I solved it in [this commit](https://github.com/n8n-io/n8n/pull/4875/commits/5f7beb7f362a79935b1ce319e84c44a80a6dd360).
I guess this is interesting only when the component is mounted.
If the license data related request is failing we don't show any content just the notification.

However, I'm still worried a bit about what the user should do in this case.